packages/ubuntu_init/lib/src/services/provd_identity_service.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:provd_client/provd_client.dart';
+import 'package:ubuntu_provision/ubuntu_provision.dart';
+
+class ProvdIdentityService implements IdentityService {
+  ProvdIdentityService({@visibleForTesting ProvdUserClient? userClient})
+      : _userClient = userClient ?? ProvdUserClient(_socketAddress, 443);
+
+  static final _socketAddress = InternetAddress(
+    '/run/provd/provd.sock',
+    type: InternetAddressType.unix,
+  );
+
+  final ProvdUserClient _userClient;
+  Identity _identity = const Identity();
+
+  @override
+  Future<Identity> getIdentity() async => _identity;
+
+  @override
+  Future<void> setIdentity(Identity identity) async {
+    if (_identity == identity) return;
+    _identity = identity;
+    return apply();
+  }
```

Comment:
See comment at the beginning. If we want to maintain a state here, we should call `apply` separately at a later point after which we can't return to the identity page in the UI.